### PR TITLE
Implement core estate catalogue model, ingestion tooling and schema

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds a new estate catalogue system under ghillie/catalogue: typed models, YAML loader, validation rules, JSON Schema generation, a CLI, and tests. Includes a first-pass ingestion surface for catalogue definitions and a ready-to-validate JSON Schema artifact.

## Changes
- New catalogue package under ghillie/catalogue with:
  - __init__.py to export types and helpers
  - models.py: Typed structures for Catalogue, Programme, Project, Component, Repository, ComponentLink, NoiseFilters, StatusSettings, etc.
  - loader.py: YAML 1.2 compliant loader and validation entry points
  - cli.py: CLI to lint a catalogue and export schema/json artefacts
  - validation.py: Validation rules and error types for catalogue integrity
  - schema.py: JSON Schema generation for catalogue via msgspec
- JSON Schema artifact schemas/catalogue.schema.json generated for validation
- Example catalogue: examples/wildside-catalogue.yaml showcasing multi-repo scenarios
- Tests:
  - Unit tests for catalogue schema validation (tests/unit/test_catalogue_schema.py)
  - Behavioural tests for catalogue ingestion/validation (tests/features/catalogue_ingestion.feature and steps)
- Updated pyproject.toml to include dependencies needed for catalogue tooling (msgspec, ruamel.yaml) and test tooling
- New/updated docs and examples to reflect catalogue tooling usage

## How to use
- Validate a catalogue and export artifacts:
  python -m ghillie.catalogue.cli examples/wildside-catalogue.yaml \
    --schema-out schemas/catalogue.schema.json \
    --json-out .cache/catalogue.json

- Validate against the JSON Schema with pajv (assumes pajv is installed):
  pajv -s schemas/catalogue.schema.json -d .cache/catalogue.json

- Load and lint a catalogue from Python:
  from ghillie.catalogue import lint_catalogue
  catalogue = lint_catalogue('examples/wildside-catalogue.yaml')

## Testing plan
- Unit tests cover schema generation and basic validation scenarios
- Feature tests exercise end-to-end catalogue ingestion, including:
  - ensuring planned components without repositories are surfaced
  - validating cross-component dependencies
  - validating the JSON Schema via pajv
- Tests rely on pytest, pytest-bdd (BDD steps), and pajv for schema validation
- Ensure pajv is available in the environment when running feature tests

## Files touched / added
- ghillie/catalogue/__init__.py (new)
- ghillie/catalogue/cli.py (new)
- ghillie/catalogue/loader.py (new)
- ghillie/catalogue/models.py (new)
- ghillie/catalogue/schema.py (new)
- ghillie/catalogue/validation.py (new)
- schemas/catalogue.schema.json (new)
- examples/wildside-catalogue.yaml (new)
- tests/features/catalogue_ingestion.feature (new)
- tests/features/steps/test_catalogue_steps.py (new)
- tests/unit/test_catalogue_schema.py (new)
- pyproject.toml (updated with catalogue dependencies)

## Notes
- This is Phase 1: introduction to the catalogue surface, focusing on modelling and validation. Ingestion, storage, and downstream usage can now rely on a validated, versioned catalogue as a source of truth.
- No existing public APIs are affected; catalogue tooling is additive and optional.

## Migration / upgrade considerations
- None required for existing code paths; catalogue module is self-contained and additive.
- If you adopt the catalogue in scripts or pipelines, you may start by validating and exporting the JSON artefact to ensure downstream compatibility.

## Next steps
- Integrate catalogue-driven ingestion into workflows (estate provisioning, etc.).
- Expand tests for edge cases (duplicate keys, invalid relationships, complex dependency graphs).
- Consider additional catalogue extensions (more relationship kinds, versioning metadata, and importers for other sources).

🌿 Generated by Terry

📎 **Task**: https://www.terragonlabs.com/task/d4a48f0c-27fa-457a-8ab6-1e7e2c670eda

### CODE REVIEW

File: ghillie/catalogue/validation.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    project_key: str,
+    component: Component,
+    state: ValidationState,
+) -> None:
+    if state.known_components is None:
+        state.issues.append(
+            "internal error: known_components not populated before relationship "
+            "validation"
+        )
+        return
```
[the remark this answers is not in the document]
Comment (leynos) [reply ghillie/catalogue/validation.py]: Has this now been resolved?

File: ghillie/catalogue/schema.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+from .models import Catalogue
+
+SCHEMA_ID = "https://ghillie.example/schemas/catalogue.json"
+
+
+def build_catalogue_schema() -> dict[str, typ.Any]:
+    """Return the JSON Schema describing the catalogue structures."""
+    schema = msgspec.json.schema(Catalogue)
+    schema.setdefault("$id", SCHEMA_ID)
```
[the remark this answers is not in the document]
Comment (leynos) [reply ghillie/catalogue/schema.py]: Has this now been resolved?

File: tests/unit/test_catalogue_schema.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      summarise_dependency_prs: false
+        """,
+        encoding="utf-8",
+    )
+
+    with pytest.raises(CatalogueValidationError) as excinfo:
+        lint_catalogue(catalogue_file)
+
+    assert "missing-component" in str(excinfo.value)
+    assert "alpha-worker" in str(excinfo.value)
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/unit/test_catalogue_schema.py]: Has this now been resolved?

[the remark this answers is not in the document]
Comment (leynos) [reply tests/unit/test_catalogue_schema.py]: Has this now been resolved?

File: tests/features/steps/test_catalogue_steps.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    catalogue: Catalogue = context["catalogue"]
+    wildside = next(
+        project for project in catalogue.projects if project.key == "wildside"
+    )
+    component = next(
+        comp for comp in wildside.components if comp.key == "wildside-ingestion"
+    )
+
+    assert component.repository is None
+    assert component.lifecycle == "planned"
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/features/steps/test_catalogue_steps.py]: Has this now been resolved?

File: docs/ghillie-proposal.md [hunk 5]
```diff
@@ -1,8 +1,13 @@
-Here’s a design for an “estate-level” status system that actually understands what’s going on, not just counts commits.
+# Ghillie proposal
 
-I’ll go top‑down, then zoom into the bits that matter: repo relationships, multi-repo projects, previous status awareness, and where the giant context windows actually help.
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/ghillie-proposal.md]: Has this now been resolved?

File: docs/ghillie-proposal.md [hunk 6]
```diff
@@ -310,19 +340,22 @@ For each repo + time window, build an “evidence bundle”, e.g.:
 
 Compress noisy details before hitting the model. For instance:
 
-- Instead of full diffs, give: “Modified files: [list of paths]” and a very short summarised description of changes (which you can get via a first small LLM call if you like).
+- Instead of full diffs, give: “Modified files: [list of paths]” and a very
+  short summarised description of changes (which the system can get via a first
```
[the remark this answers is not in the document]
Comment (leynos) [reply docs/ghillie-proposal.md]: @coderabbitai "summarize" is correct for en-GB-oxendict (-ize / -yse / -our)

File: ghillie/catalogue/cli.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    if args.json_out:
+        args.json_out.write_bytes(msgspec.json.encode(catalogue))
+
+    print(
+        f"catalogue {catalogue_path} is valid "
+        f"({len(catalogue.projects)} projects / "
+        f"{sum(len(project.components) for project in catalogue.projects)} components)"
+    )
+    return 0
```
[the remark this answers is not in the document]
Comment (leynos) [reply ghillie/catalogue/cli.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.
